packages/backend/src/managers/applicationManager.ts

```diff
@@ -133,7 +133,13 @@ export class ApplicationManager extends Publisher<ApplicationState[]> {
       const configAndFilteredContainers = this.getConfigAndFilterContainers(recipe.config, localFolder);
 
       // get model by downloading it or retrieving locally
-      const modelPath = await this.modelsManager.requestDownloadModel(model, {
+      let modelPath = await this.modelsManager.requestDownloadModel(model, {
+        'recipe-id': recipe.id,
+        'model-id': model.id,
+      });
+
+      // upload model to podman machine if user system is supported
+      modelPath = await this.modelsManager.uploadModelToPodmanMachine(model, modelPath, {
```

Comment:
We might want a fallback in case it failed ? Like using the default mounting ? let's say for XYZ reasons, it failed ?

Reply:
when i talked with Jeff about adding an option to allow users decide to upload or not the model, he said that he didn't want it, so the upload should be the default behavior, so i thought that if the upload fails the whole workflow fails. Like if the download step fails. But it can be changed, yes